src/deploymentTemplateCodeLenses.ts

```diff
@@ -78,18 +78,17 @@ export class ParameterDefinitionCodeLens extends ResolvableCodeLens {
             assert(associatedDocument instanceof DeploymentParameters);
             const dp = associatedDocument as DeploymentParameters;
 
-            const paramValue = dp.getParameterValue(this.parameterDefinition.nameValue.unquotedValue)
-                ?.value;
+            const param = dp.getParameterValue(this.parameterDefinition.nameValue.unquotedValue);
+            const paramValue = param?.value;
+            const paramReference = param?.reference;
             const givenValueAsString = paramValue?.toFullFriendlyString();
             const defaultValueAsString = this.parameterDefinition.defaultValue?.toFullFriendlyString();
 
             let title;
-            if (givenValueAsString !== undefined) {
-                if (this.isKeyVaultReference(paramValue)) {
-                    title = 'Value: (KeyVault reference)';
-                } else {
-                    title = `Value: ${givenValueAsString}`;
-                }
+            if (!!paramReference) {
```

Comment:
>paramReference [](start = 18, length = 14)

I'm a little confused why we aren't looking for keyVault and id properties inside of a reference like we did before. Even if the structure changed, is it enough that there is a reference property in the parameter value? It would probably take me days to fully understand this, so I guess I'll trust you on this one. Doesn't seem like the code is doing what the PR description said it's doing.

Reply:
Don't know for sure, frankly.  But doesn't really seem to make sense to be overly picky about it.  It's definitely not a value, but also definitely not "value not found", which is what it would be otherwise.